engine/schema/src/main/java/org/apache/cloudstack/acl/dao/ProjectRoleDaoImpl.java

```diff
@@ -17,45 +17,56 @@
 
 package org.apache.cloudstack.acl.dao;
 
-import java.util.List;
-
-import org.apache.cloudstack.acl.ProjectRoleVO;
-
 import com.cloud.utils.db.GenericDaoBase;
 import com.cloud.utils.db.SearchBuilder;
 import com.cloud.utils.db.SearchCriteria;
+import org.apache.cloudstack.acl.ProjectRoleVO;
 import org.apache.commons.lang3.StringUtils;
 
+import java.util.List;
+
 public class ProjectRoleDaoImpl extends GenericDaoBase<ProjectRoleVO, Long>  implements ProjectRoleDao{
     private final SearchBuilder<ProjectRoleVO>  ProjectRoleSearch;
 
     public ProjectRoleDaoImpl() {
         super();
 
         ProjectRoleSearch = createSearchBuilder();
-        ProjectRoleSearch.and("name", ProjectRoleSearch.entity().getName(), SearchCriteria.Op.LIKE);
+        ProjectRoleSearch.and("name", ProjectRoleSearch.entity().getName(), SearchCriteria.Op.EQ);
         ProjectRoleSearch.and("project_id", ProjectRoleSearch.entity().getProjectId(), SearchCriteria.Op.EQ);
+        ProjectRoleSearch.and("keyword", ProjectRoleSearch.entity().getName(), SearchCriteria.Op.LIKE);
         ProjectRoleSearch.done();
 
     }
     @Override
-    public List<ProjectRoleVO> findByName(String name, Long projectId) {
+    public List<ProjectRoleVO> findByName(String name, Long projectId, String keyword) {
```

Comment:
this looks to accept both name and keyword and search for name adhering to both. Is that intended?

Reply:
Name would take precedence @DaanHoogland , so it's effectively not going to use the keyword passes. I can actually remove it.

Reply:
@DaanHoogland 
yes, in most list apis, keyword is acceptable with other parameters like name, type, etc.

for this particular api, I personally think it is not necessary to accept `keyword`
but, it is ok to me.